distgo/config/internal/v0/configproduct.go

```diff
@@ -20,6 +20,10 @@ import (
 
 // ProductConfig represents user-specified configuration on how to build a specific product.
 type ProductConfig struct {
+	// ProductName specifies the product name for the product. If not specified,
+	// the product ID will generally be used.
+	ProductName *string `yaml:"product-name,omitempty"`
+
```

Comment:
For consistency, I would like this to resemble https://github.com/palantir/distgo/blob/7827e2e713023fb19b1b12fa98c6244036acd117/distgo/config/internal/v0/configbuild.go#L22-L28 more closely (although it doesn't need to be an actual template):

```suggestion
	// Name is the name of the product. This value should be used by tasks that need to use
	// a product name in their rendered configuration or output.
	//
	// If a value is not specified, the value of ProductID is used as the default value.
	Name *string `yaml:"name,omitempty"`
```